packages/headless/src/features/commerce/context/context-validation.ts

```diff
@@ -1,10 +1,16 @@
-import {RecordValue, Schema} from '@coveo/bueno';
+import {RecordValue, Schema, StringValue} from '@coveo/bueno';
+import {CurrencyCodeISO4217} from '@coveo/relay-event-types';
 import {
   nonEmptyString,
   nonRequiredEmptyAllowedString,
   requiredNonEmptyString,
 } from '../../../utils/validate-payload';
 
+export const currencyDefinition = new StringValue<CurrencyCodeISO4217>({
```

Comment:
Important to note that this definition won't evaluate in the runtime if the string set is actually a currency. Currently, we don't have an exposed list of values that could be set in the `constrainTo` parameter. 

The question would be is if it's worth adding a list in this project (which would increase the weight of the project) and if the runtime validation would bring value in that case.

Reply:
I've done a bit of digging, and I think we could get that list from https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/supportedValuesOf

```javascript
Intl.supportedValuesOf('currency')
```

So, no need to define that list somewhere else and import it, it seems native to all modern browser and node environments

Reply:
Oh, TIL! That's interesting to know :eyes:. Will try to use that.

Reply:
Would `Intl` be available in nodejs when rendering on the server? Could be something to consider in the cost of validating in headless.

Also worth noting that both Commerce API and and Event API will soon be validating the value at runtime. If the value is invalid, Commerce API will not return results,

Reply:
> Would `Intl` be available in nodejs when rendering on the server? Could be something to consider in the cost of validating in headless.
> 
> Also worth noting that both Commerce API and and Event API will soon be validating the value at runtime. If the value is invalid, Commerce API will not return results,

I had the same question for Oli earlier, short answer yes.
Long answer: Yes, on Node 18+, which is the minimal version of NodeJS we will support for V3 (so essentially that we will support for this engine once GA)
![image](https://github.com/coveo/ui-kit/assets/12366410/ace9ef89-75c9-4af2-a215-59b7a0983d68)

Reply:
Just FYI, if I try to add this function in the code, I will get this error: `property 'supportedvaluesof' does not exist on type 'typeof intl'`

From my understanding, we have to use at least Typescript 5.1, Nodejs v18 and use at least "ES2022" lib (in the tsconfig). There is a workaround describe [here](https://github.com/microsoft/TypeScript/issues/49231), but I wonder what would be the best approach to use this method.

Reply:
> Just FYI, if I try to add this function in the code, I will get this error: `property 'supportedvaluesof' does not exist on type 'typeof intl'`
> 
> From my understanding, we have to use at least Typescript 5.1, Nodejs v18 and use at least "ES2022" lib (in the tsconfig). There is a workaround describe [here](https://github.com/microsoft/TypeScript/issues/49231), but I wonder what would be the best approach to use this method.

Can you try to add `"ES2022.Intl"` into `lib` array of the tsconfig of headless?
I think it's likely to work.

Reply:
Yep it works that way!

I think I still need to do casting at some point though (Relay expect `CurrencyCodeISO4217`, so even if we constrain to the values from Intl.supportedValuesOf('currency'), the type is considered as string[]). 

On another note, there is some differences between the values returned by `Intl` and `CurrencyCodeISO4217`; the latter contains crypto currencies (such as bitcoin).